packages/protocol/contracts/tokenvault/adaptors/USDCAdaptor.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+//  _____     _ _         _         _
+// |_   _|_ _(_) |_____  | |   __ _| |__ ___
+//   | |/ _` | | / / _ \ | |__/ _` | '_ (_-<
+//   |_|\__,_|_|_\_\___/ |____\__,_|_.__/__/
+
+pragma solidity ^0.8.20;
+
+import "../../common/EssentialContract.sol";
+import "../IBridgedERC20.sol";
+
+interface IUSDC {
+    function burn(uint256 amount) external;
+    function mint(address to, uint256 amount) external;
+    function transferFrom(address from, address to, uint256 amount) external;
+}
+
+/// @title USDCAdaptor
+contract USDCAdaptor is EssentialContract, IBridgedERC20 {
+    IUSDC public usdc;
+    address public migratingFrom;
+    uint256[48] private __gap;
+
+    error INVALID_PARAMS();
+    error PERMISSION_DENIED();
+    error UNSUPPORTED_OP();
+
+    function init(address _adressManager, IUSDC _usdc) external initializer {
+        _Essential_init(_adressManager);
+        usdc = _usdc;
+    }
+
+    function mint(address account, uint256 amount) external nonReentrant whenNotPaused {
+        if (msg.sender != resolve("erc20_vault", true) && msg.sender != migratingFrom) {
+            revert PERMISSION_DENIED();
+        }
+
+        usdc.mint(account, amount);
+    }
+
+    /// @dev Warning:
+    /// 1) Users must set up allowances for this adaptor so USDC can be transferred from users
+    /// addresses to the adaptor.
+    /// 2) This adaptor must be granted the correct role by the USDC contract in order to burn
+    /// native USDC tokens.
```

Comment:
I think it is very convenient (for the Bridge team) to ask for `approve()` on the `ERC20Vault` - in general. If we introduce deployable, standalone contracts (like this), it might make their life easier to always query which contract to be 'approved' - especially if we will see lots of native support.

So what i'd do (and i know it only bc. Brecht advised me on the other branch) is exactly the same as you doing/implementing here, but this contract could be an "implementation only" contract, without being `EssentialContract`. -> A "stateless" delegate contract.

And the ERC20Vault simply would delegatecall into this contract - so basically this contract just an "extension" of that ERC20Vault - and acting on behalf of that.
See here:
https://github.com/taikoxyz/taiko-mono/blob/081c1b9ac23ac287d0ba06fa9e9474780c26a5b8/packages/protocol/contracts/tokenvault/ERC20Vault.sol#L405